### PR TITLE
Fix(DB) Preachers path.

### PR DESCRIPTION
**Pull request by Rorschach91**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [X] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes 

## SOURCE:

The changes have been validated through:
- [X] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [X] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [X] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [X] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. Create a DK.
2. Use .mod phase 2.
3. Go to New Avalon.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/Scarlet_Preachers_path.sql
Comment (elthehablo) [inline data/sql/updates/pending_db_world/Scarlet_Preachers_path.sql:1]:
```sql
SELECT COLUMN_NAME, DATA_TYPE FROM INFORMATION_SCHEMA.COLUMNS WHERE table_name = 'waypoint_data';
```

image

It's best to insert values as the datatypes they should be and not as strings. I think it's possibly harmless here, but it's convention and good practice to stick to those datatypes

**Review by elthehablo — CHANGES_REQUESTED** [review]

**Review by elthehablo — APPROVED** [review]

**Comment by Rorschach91** [comment]
@heyitsbench Are there sniff regarding these npcs?

**Review by Helias — APPROVED** [review]

**Comment by heyitsbench** [comment]
> heyitsbench Are there sniffs regarding these npcs?

Actively collecting them, these creatures are tricky since they share spawns with other creature entries.

File: data/sql/updates/pending_db_world/Scarlet_Preachers_path.sql [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+UPDATE `creature` SET `MovementType`= 2 WHERE `guid` = 129170 AND `id1` = 28594;
+UPDATE `creature` SET `MovementType`= 2 WHERE `guid` = 129171 AND `id1` = 28594;
+
+
+DELETE FROM `creature_addon` WHERE (`guid` IN (129165, 129168, 129170, 129171));
+INSERT INTO `creature_addon` (`guid`, `path_id`, `mount`, `bytes1`, `bytes2`, `emote`, `visibilityDistanceType`, `auras`) VALUES
+(129165, 12916500, 0, 0, 0, 0, 0, NULL),
+(129168, 12916800, 0, 0, 0, 0, 0, NULL),
+(129170, 12917000, 0, 0, 0, 0, 0, NULL),
+(129171, 12917100, 0, 0, 0, 0, 0, NULL);
```
Comment (heyitsbench) [inline data/sql/updates/pending_db_world/Scarlet_Preachers_path.sql:91]: These two do not exist on Cata Classic, at least not in the same phase as the others I've observed.

Comment (Rorschach91) [reply data/sql/updates/pending_db_world/Scarlet_Preachers_path.sql:91]: I cannot enter on Cata classic PTR (servers are off-line). 
Currently I can only confirm that they are present on retail:

World Of Warcraft Screenshot 2024 10 15 - 14 24 07 81

Comment (Rorschach91) [reply data/sql/updates/pending_db_world/Scarlet_Preachers_path.sql:91]: Edit: they are also present on Cata Classic: [Video](https://www.youtube.com/watch?v=4teUyejpjjw) from 1:40.

Comment (heyitsbench) [reply data/sql/updates/pending_db_world/Scarlet_Preachers_path.sql:91]: It is a phase issue, I was in the initial assault on New Avalon phase, they seem to only be in this Eye of Acherus phase.

Comment (Rorschach91) [reply data/sql/updates/pending_db_world/Scarlet_Preachers_path.sql:91]: On Ac these Preachers already are in Eye of Acherus phase.

Comment (Rorschach91) [reply data/sql/updates/pending_db_world/Scarlet_Preachers_path.sql:91]: Oh, ok. Now it is clear for me :)

Comment (heyitsbench) [reply data/sql/updates/pending_db_world/Scarlet_Preachers_path.sql:91]: https://gist.github.com/heyitsbench/cc972bf4ae11eefa3605a10cd2d0a049
Here's the sniffed data. It appears the two patrolling in front of the chapel are in a formation, i.e. one in front, one following. In the sniffed data, the leader is 0x28594A (fake GUID), and the follower is 0x28594B. My first time even looking at creature formations, so give me a sec to figure that one out. 😛

Comment (Rorschach91) [reply data/sql/updates/pending_db_world/Scarlet_Preachers_path.sql:91]: > https://gist.github.com/heyitsbench/cc972bf4ae11eefa3605a10cd2d0a049 Here's the sniffed data. It appears the two patrolling in front of the chapel are in a formation, i.e. one in front, one following. In the sniffed data, the leader is 0x28594A (fake GUID), and the follower is 0x28594B. My first time even looking at creature formations, so give me a sec to figure that one out. 😛

Eh, same for me. I'm currently learning creature formation for another group inside New Avalon (Eye of Acherus phase), but i'm blocked XD

**Comment by heyitsbench** [comment]
Why was this merged while actively being discussed?

**Comment by Helias** [comment]
> Why was this merged while actively being discussed?

I merged it by mistake 😓 , sorry

**Comment by Rorschach91** [comment]
Oh, well. We can open a new PR when the time comes anyway. It is not a problem.

**Pull request merged**